### PR TITLE
fix: [Main] Set EXTRA_PROXY_ROUTE_LIST for all add on services except Device Virtual

### LINKED ISSUE #405: Add on services are not added to API Gateway via Proxy Setup
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 Compose Builder Makefile

### Is this a regression?

 Not sure. Thought this work at one time in Levski

### Description and Minimal Reproduction [**REQUIRED**]

Run `make gen ds-onvif-camera ds-usb-camera ds-bacnet-ip ds-bacnet-mstp ds-modbus ds-mqtt ds-rest ds-snmp ds-virtual ds-llrp ds-coap ds-gpio asc-http asc-mqtt asc-sample asc-metrics as-llrp asc-ex-mqtt`

`EDGEX_ADD_PROXY_ROUTE` in generated compose file is empty. Should have proxy routes for each of the add on services.

## 🔥 Exception or Error
Not Found error for https://localhost:8443/device-gpio/api/v3/ping and others above (except device-virtual)

## 🌍  Your Environment
N/A

### PR DESCRIPTION
**Pull request by lenny-intel**
fixes #405

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A*
  <link to docs PR>

## Testing Instructions
From compose builder run `make run ds-onvif-camera ds-usb-camera ds-bacnet-ip ds-modbus ds-mqtt ds-rest ds-snmp ds-virtual ds-llrp ds-coap ds-gpio asc-http asc-mqtt asc-sample asc-metrics as-llrp asc-ex-mqtt mqtt-broker`
For each of the above services (expect mqtt-broker) verify `https://localhost:8443/<service-key>/api/v3/ping` results in 200 response.

### CODE REVIEW

**Review by bnevis-i — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
@cherrycl , are the 2 failing tests on this PR known failures? THX!

File: compose-builder/Makefile [hunk 1]
```diff
@@ -126,6 +126,12 @@ ifeq (ds-onvif-camera, $(filter ds-onvif-camera,$(ARGS)))
 		else
 			KNOWN_SECRETS_LIST:=$(KNOWN_SECRETS_LIST),redisdb[device-onvif-camera],message-bus[device-onvif-camera]
 		endif
+		PROXY_ROUTE:=device-onvif-camera.http://edgex-device-onvif-camera:59984
+		ifeq ($(EXTRA_PROXY_ROUTE_LIST),)
+			EXTRA_PROXY_ROUTE_LIST:=$(PROXY_ROUTE)
+		else
+			EXTRA_PROXY_ROUTE_LIST:=$(EXTRA_PROXY_ROUTE_LIST),$(PROXY_ROUTE)
+		endif
```
Comment (jim-wang-intel) [inline compose-builder/Makefile:134]: shall we absorb this into the shell script so there is no cloggier in the Make file?

Comment (lenny-intel) [reply compose-builder/Makefile:134]: Yea, I already have an issue for that, but best to hold off until next release.
https://github.com/edgexfoundry/edgex-compose/issues/284

Comment (jim-wang-intel) [reply compose-builder/Makefile:134]: ok

**Review by jim-wang-intel — COMMENTED** [review]
@lenny-intel  do we apply this ADD Extra route logic into taf-secty ?

**Comment by lenny-intel** [comment]
> do we apply this ADD Extra route logic into taf-secty ?

TAF team already added it where they needed it.

**Comment by cherrycl** [comment]
recheck

**Pull request merged**